test/integration/web_console_access_test.go

```diff
@@ -99,6 +119,73 @@ func TestAccessDisabledWebConsole(t *testing.T) {
 		"console/java":        {http.StatusForbidden, ""},
 	} {
 		url := masterOptions.AssetConfig.MasterPublicURL + "/" + endpoint
-		tryAccessURL(t, url, exp.statusCode, exp.location)
+		tryAccessURL(t, url, exp.statusCode, exp.location, nil)
+	}
+}
+
+func TestAccessOriginWebConsoleMultipleIdentityProviders(t *testing.T) {
+	masterOptions, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Replace the default IdentityProvider with an AllowAll provider
+	masterOptions.OAuthConfig.IdentityProviders[0] = configapi.IdentityProvider{
+		Name:            "foo",
+		UseAsChallenger: true,
+		UseAsLogin:      true,
+		MappingMethod:   "claim",
+		Provider:        &configapi.AllowAllPasswordIdentityProvider{},
+	}
+
+	// Set up a second AllowAll provider
+	masterOptions.OAuthConfig.IdentityProviders = append(masterOptions.OAuthConfig.IdentityProviders, configapi.IdentityProvider{
+		Name:            "bar",
+		UseAsChallenger: true,
+		UseAsLogin:      true,
+		MappingMethod:   "claim",
+		Provider:        &configapi.AllowAllPasswordIdentityProvider{},
+	})
+
+	// Launch the configured server
+	if _, err = testserver.StartConfiguredMaster(masterOptions); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Create a map of URLs to test
+	type urlResults struct {
+		statusCode int
+		location   string
+	}
+
+	urlMap := make(map[string]urlResults)
+	linkRegexps := make([]string, 0)
+
+	// Verify that the plain /login URI is unavailable when multiple IDPs
+	// are in use.
+	urlMap["login"] = urlResults{http.StatusForbidden, ""}
```

Comment:
bake the leading slash into the keys and `loginSelectorBase` instead of in the append